components/net/http_loader.rs

```diff
@@ -2139,11 +2139,22 @@ async fn cors_preflight_fetch(
     // Step 4
     let headers = get_cors_unsafe_header_names(&request.headers);
 
-    // Step 5
+    // Step 5 If headers is not empty, then:
     if !headers.is_empty() {
-        preflight
-            .headers
-            .typed_insert(AccessControlRequestHeaders::from_iter(headers));
+        // 5.1 Let value be the items in headers separated from each other by `,`
+        // TODO(36451): replace this with typed_insert when headers fixes headers#207
+        preflight.headers.insert(
+            ACCESS_CONTROL_REQUEST_HEADERS,
+            HeaderValue::from_bytes(
+                headers
+                    .iter()
+                    .map(|header| header.to_string())
+                    .reduce(|acc, x| acc + "," + &x)
```

Comment:
This is a bit naive overall, not sure if its going to ever be a performance problem though.

Reply:
Definitely not something to attempt to optimize prematurely!

Reply:
This could be optimized a bit and be a little bit less verbose using `itertools::join`, I think, which we use pretty consistently in `script`. Do you mind making that change?